src/App.tsx

```diff
@@ -15,47 +15,100 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE = 0,
+  ALPHABET = 1,
+  LENGTH = 2,
+}
+
+function getReorderedGoods(
+  goods: string[],
+  isReversed: boolean,
+  sortType: SortType,
+):string[] {
+  const visible = [...goods];
+
+  visible.sort((goodFirst, goodSecond) => {
+    switch (sortType) {
+      case SortType.ALPHABET:
+        return goodFirst.localeCompare(goodSecond);
+      case SortType.LENGTH:
+        return goodFirst.length - goodSecond.length;
+      case SortType.NONE:
+        return 0;
+      default:
+        return 0;
+    }
+  });
+
+  if (isReversed) {
+    visible.reverse();
+  }
+
+  return visible;
+}
+
 export const App: React.FC = () => {
+  const copiedData = [...goodsFromServer];
+  const [isReversed, setIsReversed] = useState(false);
+  const [sortType, setSortType] = useState(SortType.NONE);
+
+  const reverse = () => {
+    setIsReversed(prevState => !prevState);
+  };
+
+  const changedArr: string[]
+  = useMemo(() => getReorderedGoods(copiedData, isReversed, sortType),
+    [copiedData, isReversed, sortType]);
```

Comment:
Check the code style, It's not easy to read it now. BTW you don't really need to specify the variable type here.
```suggestion
  const changedArr: string[] = useMemo(
    () => getReorderedGoods(copiedData, isReversed, sortType),
    [copiedData, isReversed, sortType]
  );
```